### PR TITLE
Three.js : Add WebGLInfo.

### PR DESCRIPTION
**Pull request by Astrak**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [] Add or edit tests to reflect the change. (Run with `npm test`.) (test were run but none added, three.js's core is nearly not tested).
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: WebGLInfo has been created in r91 : https://github.com/mrdoob/three.js/commits/dev/src/renderers/webgl/WebGLInfo.js
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/three/three-core.d.ts [hunk 1]
```diff
@@ -6065,6 +6051,26 @@ export class WebGLLights {
...
+ * An object with a series of statistical information about the graphics board memory and the rendering process. 
+ */ 
+export class WebGLInfo { 
+    autoReset: boolean; 
+    memory: { 
+        geometries: number; 
+        textures: number; 
+    }; 
+    programs: number | null; 
+    render: {
```
Comment (efokschaner) [inline types/three/three-core.d.ts]: render appears to have a `frame` member. https://github.com/mrdoob/three.js/blob/dev/src/renderers/webgl/WebGLInfo.js#L13

File: types/three/three-core.d.ts [hunk 2]
```diff
@@ -6065,6 +6051,26 @@ export class WebGLLights {
...
+    }; 
+    programs: number | null; 
+    render: { 
+        calls: number; 
+        lines: number; 
+        points: number; 
+        triangles: number; 
+    }; 
+    reset(): void; 
+    update(): void;
```
Comment (efokschaner) [inline types/three/three-core.d.ts]: update takes params, but is this part of three's intentionally exposed API? Is it documented?

**Review by efokschaner — CHANGES_REQUESTED** [review]
Thanks for the improvements. I did leave some suggestions.

File: types/three/three-core.d.ts [hunk 3]
```diff
@@ -6065,6 +6051,26 @@ export class WebGLLights {
...
+/** 
+ * An object with a series of statistical information about the graphics board memory and the rendering process. 
+ */ 
+export class WebGLInfo { 
+    autoReset: boolean; 
+    memory: { 
+        geometries: number; 
+        textures: number; 
+    }; 
+    programs: number | null;
```
Comment (efokschaner) [inline types/three/three-core.d.ts]: I don't think this is the right type for `programs` it appears to be set here: https://github.com/mrdoob/three.js/blob/bd63a6cc8dd7fe11b530410fab9f67a9de359be8/src/renderers/WebGLRenderer.js#L276

**Review by efokschaner — CHANGES_REQUESTED** [review]

File: types/three/three-core.d.ts (hunk 2 above)
Comment (Astrak) [reply types/three/three-core.d.ts]: Correct... well it is exposed with the rest of the object but it's only supposed to be used internally. Removed from the types.

**Review by efokschaner — APPROVED** [review]
Thanks!

**Comment by rbuckton** [comment]
Thanks for the contribution!

**Pull request merged**